### PR TITLE
Replace "org/repo" string with struct

### PR DESCRIPTION
**Pull request by matthyx**
Relates to #16223

This is the first commit of the work:
- add new `Repo` struct
- use `Repo` in helpProvider

Subsequently I will replace "org/repo" in other places, but I wanted to keep this PR manageable.

/cc @alvaroaleman

### CODE REVIEW

File: prow/pluginhelp/hook/hook.go [hunk 1]
```diff
@@ -83,7 +83,12 @@ func (ha *HelpAgent) generateNormalPluginHelp(config *plugins.Configuration, rev
 			ha.log.Warnf("No help is provided for plugin %q.", name)
 			continue
 		}
-		help, err := provider(config, revMap[name])
+		var enabledRepos []plugins.Repo
+		for _, repo := range revMap[name] {
```
Comment (alvaroaleman) [inline prow/pluginhelp/hook/hook.go]: Whats the reason for not using the `Repo` struct as values in the `revMap`?

Comment (matthyx) [reply prow/pluginhelp/hook/hook.go]: I just wanted to focus on the help provider in this PR. Then I will refine in specific areas like here and some parts of Config

**Pull request merged**